src/System.Runtime/tests/System/NullableMetadataTests.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System.Buffers;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace System.Runtime.Tests
+{
+    public static class NullableMetadataTests
+    {
+        private const string NullableAttributeFullName = "System.Runtime.CompilerServices.NullableAttribute";
+        private const string NullableContextAttributeFullName = "System.Runtime.CompilerServices.NullableContextAttribute";
+        private const string NullablePublicOnlyAttributeFullName = "System.Runtime.CompilerServices.NullablePublicOnlyAttribute";
+
+        private static IEnumerable<CustomAttributeData> GetNullableAttributes(this IEnumerable<CustomAttributeData> attributes) =>
+            attributes.Where(attribute => attribute.AttributeType.FullName.Equals(NullableAttributeFullName) ||
+                                          attribute.AttributeType.FullName.Equals(NullableContextAttributeFullName));
+
+        private static CustomAttributeData GetNullablePublicOnlyAttribute(this IEnumerable<CustomAttributeData> attributes) =>
+            attributes.Where(attribute => attribute.AttributeType.FullName.Equals(NullablePublicOnlyAttributeFullName)).FirstOrDefault();
+
+        private static bool IsProtected(this MemberInfo info)
+        {
+            if (info is MethodBase methodBase)
+            {
+                return methodBase.IsFamily || methodBase.IsFamilyOrAssembly;
+            }
+            else if (info is FieldInfo fieldInfo)
+            {
+                return fieldInfo.IsFamily || fieldInfo.IsFamilyOrAssembly;
+            }
+            else if (info is PropertyInfo propertyInfo)
+            {
+                return (propertyInfo.GetMethod != null && (propertyInfo.GetMethod.IsFamily || propertyInfo.GetMethod.IsFamilyOrAssembly)) ||
+                       (propertyInfo.SetMethod != null && (propertyInfo.SetMethod.IsFamily || propertyInfo.SetMethod.IsFamilyOrAssembly));
+            }
+            else if (info is TypeInfo typeInfo)
+            {
+                return typeInfo.IsNestedFamily || typeInfo.IsNestedFamORAssem;
+            }
+
+            return false;
+        }
+
+        private static IEnumerable<object[]> NullableMetadataTypesTestData()
+        {
+            yield return new object[] { typeof(string) };
+            yield return new object[] { typeof(Dictionary<,>) };
+            yield return new object[] { typeof(Uri) };
+            yield return new object[] { typeof(ConcurrentDictionary<,>) };
```

Comment:
are all these defined in corelib? should this include at least one type that's defined in corefx (but exposed through the same contract as corelib types)

Reply:
Uri is in System.Private.Uri in corefx, and ConcurrentDictionary is in System.Collections.Concurrent in corefx.

Reply:
`ConcurrentDictionary` and `Uri` both live in corefx and in a different ref assembly. I can add more to this list if needed.